refgenconf/refgenconf.py

```diff
@@ -2087,10 +2162,134 @@ def _select_genomes(self, genome=None, strict=False, order=None,
                 else:
                     missing.append(g)
             if missing:
-                _LOGGER.warning("Genomes do not include: {}".format(", ".join(missing)))
+                _LOGGER.warning(
+                    "Genomes do not include: {}".format(", ".join(missing)))
             return None if not filtered else filtered
         return genomes if not all(x in genomes for x in genome) else genome
 
+    def config_upgrade(target_version, filepath, force=False):
+        """
+            upgrade the config file to a target version, 
+            and alter genome_folder stucture 
+
+            :param str target_version: the version updated to
+            :param str filepath: file path for the current config file
+            :param bool force
+        """
+
+        # check if any genome lack of local fasta asset and not on server
+
+        def check_genome_digests(config):
+            missing_digest = []
+            for k, v in config["genomes"].items():
+                if not os.path.exists(config["genome_folder"]+'/'+k +
+                                      '/fasta/default/'+k+'.fa'):
+                    response = requests.get(
+                        "http://refgenomes.databio.org:82/v3/alias/genome_digest/"+k)
+                    if response.json()['detail']:
+                        missing_digest.append(k)
+
+            if missing_digest and not query_yes_no(
+                    "The following genome(s) would be lost due to the lack of local and remote fasta asset(s): \n"
+                    "{} \n"
+                    "Would you like to proceed?"
+                    .format(missing_digest)):
+                _LOGGER.info("Action aborted by the user. To use it, please downgrade refgenie: 'pip install \"refgenie>={},<{}\"'".
+                             format(REFGENIE_BY_CFG[str(config["config_version"])], REFGENIE_BY_CFG[str(REQ_CFG_VERSION)]))
+                return missing_digest
+        # reformat config file
+
+        def format_config(config, target_version):
+            # reformat the config file
+            for k, v in config["genomes"].items():
+                # create "aliases" section
+                v.aliases = [k]
+                # get genome digest from the server
+                response = requests.get(
+                    "http://refgenomes.databio.org:82/v3/alias/genome_digest/"+k)
+                if response.json()['detail']:
+                    # if genome asset not exist on serer, generate the digest using the local fasta asset
+                    ssc = SeqColClient({})
+                    try:
+                        d, _ = ssc.load_fasta(config["genome_folder"]+'/'+k +
```

Comment:
To get the path to the `fasta` file use the existing `refgenconf` API instead of stitching a path together. And again, `"default"` tag might not be the actual default one, you'll need to find out which tag has been set as the default one.